### PR TITLE
Use setup-ruby action instead of the docker image

### PR DESCRIPTION
**Pull request by tcbyrd**
I was able to get this configuration working. Bundle install inside Actions is actually quite fast, and the setup-ruby action has automatic caching. The first run of this on my test repo did the whole bundle install and jekyll build in ~31 seconds. Subsequent runs were ~10-12 seconds, which is actually faster than downloading our docker image:

image

This also makes all our starter workflows aligned on a model where you use the tools of the framework to build, and can optionally add github-specific plugins (like the `github-pages` gem) to gain additional functionality.

cc @parkr since you might be interested in what we're doing here.

### CODE REVIEW

**Comment by parkr** [comment]
Hi! setup-ruby has been very fast and reliable in my experience, and gets around the issue with actions Docker images where the work takes place in a container but a user expects that it happened in the VM (and, for example, a file they wrote in the container is not in the VM). You have a bit less control over things and you rely on the user setting up their Gemfile correctly to have Jekyll and all their desired plugins, so it's a bit scarier from a product development/user journey perspective to lose this. It's less management for you though since you're no longer the gate keepers to Jekyll & plug-in upgrades. Generally folks love speed though so if it's significantly faster, then dependabot + a Gemfile should keep users up to date over time!

**Comment by tcbyrd** [comment]
Yeh the context I realized you don't have here is this will live next to a set of "suggested workflows" for Pages as an example of how to get started customizing your own. The built-in Action will continue using our container.

**Comment by parkr** [comment]
Gotcha! Then big +1 to using the VM directly with setup-ruby!

**Review by JamesMGreene — APPROVED** [review]

**Comment by JamesMGreene** [comment]
@tcbyrd I resolved the conflicts and added the `--baseurl {base_path}` bit for the Jekyll build step like we discussed the other day.

File: pages/jekyll.yml [hunk 1]
```diff
@@ -29,16 +29,18 @@ jobs:
...
+          cache-version: 0 # Increment this number if you need to re-download cached gems
       - name: Setup Pages
         id: pages
         uses: paper-spa/configure-pages@main
-      - name: Build with Jekyll
-        uses: actions/jekyll-build-pages@v1
-        with:
-          source: ./
-          destination: ./_site
+      - run: bundle exec jekyll build --baseurl ${{ steps.pages.outputs.base_path }} # defaults output to '/_site'
```
Comment (tcbyrd) [inline pages/jekyll.yml:41]: @JamesMGreene To confirm, this output is set inside the configure-pages action? Do we need to guard on whether or not that output variable exists?

Comment (JamesMGreene) [reply pages/jekyll.yml:41]: Correct. That output should be guaranteed to exist unless the Action itself failed, so I think we're good? 🤔

Its value will either be `"/"` or `"/my-repo/"`.

**Pull request merged**